### PR TITLE
Add an HTTP/2 related rate limiting

### PR DESCRIPTION
**Pull request by maskit**

### CODE REVIEW

**Comment by bryancall** [comment]
#
[approve ci]

**Comment by bryancall** [comment]
[approve ci]

**Review by bryancall — APPROVED** [review]

File: proxy/http2/Http2ConnectionState.cc [hunk 1]
```diff
@@ -522,6 +522,17 @@ rcv_rst_stream_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
                       "reset frame wrong length");
   }
 
+  // Update RST_STREAM frame count per minute
+  cstate.increment_received_rst_stream_frame_count();
+  // Close this connection if its RST_STREAM frame count exceeds a limit
+  if (cstate.get_received_rst_stream_frame_count() > Http2::max_rst_stream_frames_per_minute) {
+    HTTP2_INCREMENT_THREAD_DYN_STAT(HTTP2_STAT_MAX_RST_STREAM_FRAMES_PER_MINUTE_EXCEEDED, this_ethread());
+    Http2StreamDebug(cstate.ua_session, stream_id, "Observed too frequent RST_STREAM frames: %u frames within a last minute",
+                     cstate.get_received_settings_frame_count());
```
Comment (bdgranger) [inline proxy/http2/Http2ConnectionState.cc:531]: @maskit Is this a typo or copy/paste error? Shouldn't it be get_received_rst_stream_frame_count()

Comment (maskit) [reply proxy/http2/Http2ConnectionState.cc:531]: Oh, you're right... Fortunately it's just a debug message.

Comment (maskit) [reply proxy/http2/Http2ConnectionState.cc:531]: https://github.com/apache/trafficserver/pull/10583

**Pull request merged**